src/shapes/itext.class.js

```diff
@@ -531,6 +531,7 @@
    * @returns {Promise<fabric.IText>}
    */
   fabric.IText.fromObject = function(object) {
+    object.styles = fabric.util.stylesFromArray(object.styles, object.text);
     return fabric.Object._fromObject(fabric.IText, object, { extraParam: 'text' });
```

Comment:
```suggestion
    return fabric.Object._fromObject(fabric.IText, Object.assign({}, object, {style: style}), { extraParam: 'text' });
```

Reply:
what I mean is the object returned by `stylesFromArray` is already composed of copied properties, so shouldn't it already be safe from mutation?

Reply:
I'm talking about the object passed to the fromObject function.
Test a case which you create a text object from an object and use the same object to create another text object.
It should break if you use styles, shouldn't it? The object will be mutated

Reply:
You can/should test it by comparing the styles property of the passed object before and after you initialized the text object

Reply:
dangit, you're right 😆